indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -84,10 +102,64 @@ def _query_registrations(regform):
             .options(joinedload('data').joinedload('field_data').joinedload('field')))
 
 
-def _get_visible_column_ids(items):
-    items = set(items)
-    special_cols = {'reg_date', 'state', 'price'}
-    return (items - special_cols), (items & special_cols)
+def _split_column_ids(items):
+    """Split column ids between custom and 'basic' (personal + special)."""
+    special_cols = [item for item in items if isinstance(item, basestring)]
+    return [item for item in items if item not in special_cols], list(special_cols)
+
+
+def _split_special_ids(items):
+    """Split column ids between 'DB-stored' (custom + personal) and special."""
+    special = []
+    normal = []
+
+    for item in items:
+        if item in SPECIAL_COLUMNS:
+            special.append(item)
+        else:
+            normal.append(item)
+    return normal, special
+
+
+def _get_basic_columns(form, ids):
+    """
+    Retrieve information needed for the header of "basic" columns (personal + special).
+
+    Returns a list of {'id': ..., 'caption': ...} dictionaries.
+    """
+    result = []
+    for item_id in PERSONAL_COLUMNS:
+        if item_id in ids:
+            field = RegistrationFormItem.find_one(registration_form=form,
+                                                  personal_data_type=getattr(PersonalDataType, item_id))
+            result.append({
+                'id': field.id,
+                'caption': field.title
+            })
+
+    for item_id in SPECIAL_COLUMNS:
+        if item_id in ids:
+            result.append({
+                'id': item_id,
+                'caption': SPECIAL_COLUMN_LABELS[item_id]
+            })
+    return result
+
+
+def _column_ids_to_db(form, ids):
+    """Translate string-based ids to DB-based RegistrationFormItem ids."""
+    result = []
+    for item_id in ids:
+        if isinstance(item_id, basestring):
```

Comment:
I just tested it. It can be the actual ID. The interface sends the field ID, which is a number except for personal data and "special" fields (price, etc...)